src/app/mycart/my-cart.component.ts

```diff
@@ -1,15 +1,82 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router} from '@angular/router';
+import { ProductsService } from '../products/products.service';
 
 @Component({
   selector: 'app-my-cart',
   templateUrl: './my-cart.component.html',
   styleUrls: ['./my-cart.component.css']
 })
 export class MyCartComponent implements OnInit {
+  name:string;
+  productsList:any=[];
+  myCart:any=[];
+  cartValues=[];
+  totalAmount:number=0;
```

Comment:
you needn't really specify the type here. Javascript can imply the type from the value